cmd/root.go

```diff
@@ -144,3 +138,134 @@ func initConfig() {
+		}
+	}
+
+	if utils.MayExists(utils.Flags.ManifestPath) {
+
+		var deployer = deployers.NewServiceDeployer()
+		deployer.ProjectPath = projectPath
+		deployer.ManifestPath = utils.Flags.ManifestPath
+		deployer.DeploymentPath = utils.Flags.DeploymentPath
+		// perform some quick check here.
+		go func() {
+			deployer.Check()
+		}()
+		deployer.IsDefault = utils.Flags.UseDefaults
+
+		deployer.IsInteractive = utils.Flags.UseInteractive
+
+		// master record of any dependency that has been downloaded
+		deployer.DependencyMaster = make(map[string]utils.DependencyRecord)
+
+		propPath := ""
+		if !utils.Flags.WithinOpenWhisk {
+			userHome := utils.GetHomeDirectory()
+			propPath = path.Join(userHome, ".wskprops")
+		}
+		whiskClient, clientConfig := deployers.NewWhiskClient(propPath, utils.Flags.DeploymentPath, deployer.IsInteractive)
+		deployer.Client = whiskClient
+		deployer.ClientConfig = clientConfig
+
+		err := deployer.ConstructDeploymentPlan()
+
+		if err != nil {
+			utils.Check(err)
+			return err
+		}
+
+		err = deployer.Deploy()
+		if err != nil {
+			utils.Check(err)
+			return err
+		} else {
+			return nil
+		}
+
+	} else {
+		if utils.Flags.WithinOpenWhisk {
+			utils.PrintOpenWhiskError(wski18n.T("missing manifest.yaml file"))
+			return errors.New("missing manifest.yaml file")
+		} else {
+			log.Println("missing manifest.yaml file")
+			return errors.New("missing manifest.yaml file")
+		}
+	}
+
+}
+
+func Undeploy() error {
+	// TODO: Work your own magic here
+	whisk.SetVerbose(utils.Flags.Verbose)
+
+	if utils.Flags.ManifestPath == "" {
+		if ok, _ := regexp.Match(deployers.ManifestFileNameYml, []byte(utils.Flags.ManifestPath)); ok {
+			utils.Flags.ManifestPath = path.Join(utils.Flags.ProjectPath, deployers.ManifestFileNameYml)
+		} else {
+			utils.Flags.ManifestPath = path.Join(utils.Flags.ProjectPath, deployers.ManifestFileNameYaml)
+		}
+
+	}
+
+	if utils.Flags.DeploymentPath == "" {
+		if ok, _ := regexp.Match(deployers.DeploymentFileNameYml, []byte(utils.Flags.ManifestPath)); ok {
+			utils.Flags.DeploymentPath = path.Join(utils.Flags.ProjectPath, deployers.DeploymentFileNameYml)
+		} else {
+			utils.Flags.DeploymentPath = path.Join(utils.Flags.ProjectPath, deployers.DeploymentFileNameYaml)
+		}
+
+	}
+
+	if utils.FileExists(utils.Flags.ManifestPath) {
+
+		var deployer = deployers.NewServiceDeployer()
+		deployer.ProjectPath = utils.Flags.ProjectPath
+		deployer.ManifestPath = utils.Flags.ManifestPath
+		deployer.DeploymentPath = utils.Flags.DeploymentPath
+
+		deployer.IsInteractive = utils.Flags.UseInteractive
+		deployer.IsDefault = utils.Flags.UseDefaults
+
+		userHome := utils.GetHomeDirectory()
+		propPath := path.Join(userHome, ".wskprops")
+
+		whiskClient, clientConfig := deployers.NewWhiskClient(propPath, utils.Flags.DeploymentPath, deployer.IsInteractive)
+		deployer.Client = whiskClient
+		deployer.ClientConfig = clientConfig
+
+		verifiedPlan, err := deployer.ConstructUnDeploymentPlan()
```

Comment:
@lzbj before we "deploy" when a "-p" package path is provided, we should (in verbose mode) tell the user (echo back) the path/filename we found and parsed/using to deploy.  We also need to have a clear error (with special text) that tells used "Manifest file not found at path 'x'".